### PR TITLE
Surf s4

### PR DESCRIPTION
**Pull request by SlothSan**
Hi Rayna, 

Ready for review. 

Cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,12 +196,15 @@ PUT
 - Checks if user email address exists in database. Adds test score to applicant entry in database.
 - To add an applicant's test score to the database, the request should be sent as a PUT request to /api/aptiudeScore in
 the following JSON format:
-  - `{"email": 'example@email.com, "score": 'integer'}`
+  - `{"email": 'example@email.com, "aptitude": 'integer'}`
```
Comment (norayna) [inline README.md:199]: surely the integer should be without quotes?

File: README.md [hunk 2]
```diff
@@ -196,12 +196,15 @@ PUT
 - Checks if user email address exists in database. Adds test score to applicant entry in database.
 - To add an applicant's test score to the database, the request should be sent as a PUT request to /api/aptiudeScore in
```
Comment (norayna) [inline README.md:197]: i think there's a typo in the URL right?

File: README.md (hunk 1 above)
Comment (SlothSan) [reply README.md:199]: Resolved

File: README.md (hunk 2 above)
Comment (SlothSan) [reply README.md:197]: Resolved

File: README.md [hunk 3]
```diff
@@ -196,12 +196,15 @@ PUT
...
 the following JSON format:
-  - `{"email": 'example@email.com, "score": 'integer'}`
+  - `{"email": 'example@email.com, "aptitude": 'integer'}`
 The score should be an integer 0-100.
 - Returns success true/false:
   - If email exists in the database, adds applicant test store to the database.
-    - `{"success": true, "message": "Aptitude score added successfully", "data": []}` 
+    - `{"success": true, "message": "Updated the applicants aptitude score", "data": []}` 
+  - If email exists in the database, and applicant already has a aptitude score then add the aptitude score to the assessmentNotes with a time stamp.
```
[the remark this answers is not in the document]
Comment (SlothSan) [reply README.md:204]: Resolved

File: app/dependencies.php [hunk 4]
```diff
@@ -131,6 +131,7 @@
     $container['CsvController'] = DI\factory('\Portal\Factories\Controllers\API\CsvControllerFactory');
     $container['GetEventCategoriesController'] = DI\factory('\Portal\Factories\Controllers\API\GetEventCategoriesControllerFactory');
     $container['AddAptitudeScoreController'] = DI\factory('\Portal\Factories\Controllers\API\AddAptitudeScoreControllerFactory');
+    $container['SubmitAptitudeScoreController'] =DI\factory('\Portal\Factories\Controllers\API\SubmitAptitudeScoreControllerFactory');
```
[the remark this answers is not in the document]
Comment (SlothSan) [reply app/dependencies.php:134]: Doh, yes ! 

Removed

File: src/Controllers/API/AddAptitudeScoreController.php [hunk 5]
```diff
@@ -10,61 +10,60 @@
...
-
-    /**
-     * Instantiates AddApplicantPageController.
-     *
-     * @param ApplicantModel $applicantModel userModel object dependency
-     */
+//* Instantiates AddApplicantPageController.
+//*
+//* @param ApplicantModel $applicantModel userModel object dependency
+//*/
```
Comment (norayna) [inline src/Controllers/API/AddAptitudeScoreController.php:16]: very weird comment notation

Comment (SlothSan) [reply src/Controllers/API/AddAptitudeScoreController.php:16]: Resolved.

File: src/Models/ApplicantModel.php [hunk 6]
```diff
@@ -557,13 +557,39 @@ public function getApplicantByEmail(string $email): array
...
+
+        $values = ['id' => $id];
+
+        $query = $this->db->prepare($sql);
+        $query->setFetchMode(\PDO::FETCH_ASSOC);
+        $query->execute($values);
+        return $query->fetch();
+    }
+
+    public function submitApplicantAptitudeScore(int $id, int $score): bool
```
Comment (norayna) [inline src/Models/ApplicantModel.php:572]: rename to `setAptitudeScore`

File: src/Models/ApplicantModel.php [hunk 7]
```diff
@@ -557,13 +557,39 @@ public function getApplicantByEmail(string $email): array
...
             'score' => $score,
-            'id' => $id,
+            'id' => $id
+        ];
+
+        $query = $this->db->prepare($sql);
+        return $query->execute($values);
+    }
+
+    public function submitApplicantAssessmentNotes(int $id, string $assessmentNote): bool
```
Comment (norayna) [inline src/Models/ApplicantModel.php:585]: rename to `appendToAssessmentNotes`

File: src/Models/ApplicantModel.php (hunk 6 above)
Comment (SlothSan) [reply src/Models/ApplicantModel.php:572]: Resolved.

File: src/Models/ApplicantModel.php (hunk 7 above)
Comment (SlothSan) [reply src/Models/ApplicantModel.php:585]: Resolved.

**Review by norayna — CHANGES_REQUESTED** [review]
ugh sorry

File: README.md [hunk 8]
```diff
@@ -194,14 +194,17 @@ DELETE
 
 PUT 
 - Checks if user email address exists in database. Adds test score to applicant entry in database.
-- To add an applicant's test score to the database, the request should be sent as a PUT request to /api/aptiudeScore in
+- To add an applicant's test score to the database, the request should be sent as a PUT request to /api/aptitudeScore in
 the following JSON format:
-  - `{"email": 'example@email.com, "score": 'integer'}`
+  - `{"email": 'example@email.com, "aptitude": integer}`
```
[the remark this answers is not in the document]
Comment (SlothSan) [reply README.md:199]: Resolved.

File: src/Controllers/API/AddAptitudeScoreController.php [hunk 9]
```diff
@@ -23,48 +23,49 @@ public function __construct(ApplicantModel $applicantModel)
...
-            return $this->respondWithJson($response, $data, $statusCode);
+        $aptitudeFromId = $this->applicantModel->getAptitudeScore($applicantId);
+        if ($aptitudeFromId['aptitude'] == null) {
+            $this->applicantModel->setAptitudeScore($applicantId, $aptitudeScore);
+            $responseBody['message'] = 'Updated the applicants aptitude score';
+            $responseBody['success'] = true;
+            return $this->respondWithJson($response, $responseBody, 200);
+        } else {
+            date_default_timezone_set("Europe/London");
+            $assessmentNote = 'New aptitude test attempt ' . date("d/m/y") . ' ' . date("H:i")
```
[the remark this answers is not in the document]
Comment (SlothSan) [reply src/Controllers/API/AddAptitudeScoreController.php]: Resolved

File: src/Models/ApplicantModel.php [hunk 10]
```diff
@@ -557,13 +557,39 @@ public function getApplicantByEmail(string $email): array
         return $result;
     }
 
-    public function setAptitudeScore($id, $score): bool
+    public function getAptitudeScore(int $id): array
+    {
+        $sql = 'SELECT `aptitude`, `assessmentNotes` FROM applicants_additional WHERE `id` = :id;';
```
Comment (norayna) [inline src/Models/ApplicantModel.php:562]: `applicants_additional` needs backticks

Comment (SlothSan) [reply src/Models/ApplicantModel.php]: Resolved, this was the old method that we started to use and renamed! - can tell you didn't do the code review.

**Review by norayna — APPROVED** [review]
approved, great work

**Pull request merged**